dlt/extract/utils.py

```diff
@@ -171,6 +178,57 @@ async def run() -> TDataItems:
         exhausted = True
 
 
+def wrap_parallel_iterator(
+    f: Union[Generator[TDataItems, Optional[Any], Optional[Any]], AnyFun]
+) -> Union[Generator[TDataItems, Optional[Any], Optional[Any]], AnyFun]:
+    """Wraps a generator for parallel extraction"""
+
+    def _wrapper(*args: Any, **kwargs: Any) -> Generator[TDataItems, None, None]:
+        is_generator = True
+        gen: Union[Generator[TDataItems, Optional[Any], Optional[Any]], AnyFun]
+        if callable(f):
+            if inspect.isgeneratorfunction(f):
+                gen = f(*args, **kwargs)
+            else:
+                is_generator = False
```

Comment:
why do we allow regular functions here? this will also execute them in a pool right?

Reply:
Transformes that are not generators: 
https://github.com/dlt-hub/dlt/blob/sthor%2Fparallelize-decorator/tests/pipeline/test_resources_evaluation.py#L365-L371  
So they execute in pool. Any reason not to support it?